engine/src/main/java/org/terasology/world/block/IBlockRegion.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.world.block;
+
+import org.joml.AABBf;
+import org.joml.LineSegmentf;
+import org.joml.Matrix4fc;
+import org.joml.Planef;
+import org.joml.Rayf;
+import org.joml.Spheref;
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.entitySystem.entity.EntityRef;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+public interface IBlockRegion extends Iterable<Vector3ic> {
+    public static BlockRegion invalid();
+
+    // -- CONSTRUCTORS ----------------------------------------------------------------------------------------------//
+    BlockRegion();
+    public BlockRegion(int minX, int minY, int minZ, int maxX, int maxY, int maxZ);
+    public BlockRegion(Vector3ic min, Vector3ic max);
+    public BlockRegion(int x, int y, int z);
+    public BlockRegion(Vector3ic block);
+    public BlockRegion(BlockRegion source)
+
+    // -- ITERABLE -------------------------------------------------------------------------------------------------//
+    public Iterator<Vector3ic> iterator();
```

Comment:
What kind of iteration is this?

Reply:
See https://github.com/MovingBlocks/Terasology/pull/4329#discussion_r547258485